pkgs/development/libraries/qrcodegen/default.nix

```diff
@@ -14,9 +14,13 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-aci5SFBRNRrSub4XVJ2luHNZ2pAUegjgQ6pD9kpkaTY=";
   };
 
-  preBuild = ''
-    cd c/
-  '';
+  sourceRoot = "source/c";
```

Comment:
Why?

Reply:
AFAIK it's the "proper" way to say "I am not building from the root of the repo" :shrug:

Reply:
I am not comfortable with this, because it relies on an undocumented feature of `fetchFromGitHub`, namely, using `source` as default `name` for the extracted directory. 
And worse, this default can be modified by the user of `fetchFromGitHub`.

Reply:
That's actually... [documented](https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/stdenv.chapter.md#sourceroot-var-stdenv-sourceroot)?

> By default the `sourceRoot` is set to `"source"`. If you want to point to a sub-directory inside your project, you therefore need to set `sourceRoot = "source/my-sub-directory"`.

Reply:
Huh o.o

It is not confirmed by practice. When unpacking a `fetchurl`'d tarball, it generates something like pname-version.

Reply:
> > By default the `sourceRoot` is set to `"source"`. If you want to point to a sub-directory inside your project, you therefore need to set `sourceRoot = "source/my-sub-directory"`.

What are the cases in which this default is modified?